SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -553,10 +560,20 @@ bool TryLoadExistingSchemaDatabase()
 			return false;
 		}
 
-		ClassPathToSchemaData = SchemaDatabase->ClassPathToSchema;
+		ActorClassPathToSchema = SchemaDatabase->ActorClassPathToSchema;
+		SubobjectClassPathToSchema = SchemaDatabase->SubobjectClassPathToSchema;
 		LevelComponentIds = SchemaDatabase->LevelComponentIds;
 		LevelPathToComponentId = SchemaDatabase->LevelPathToComponentId;
 		NextAvailableComponentId = SchemaDatabase->NextAvailableComponentId;
+
+		// Component Id generation was updated to be non-destructive, if we detect an old schema database, delete it.
+		if (ActorClassPathToSchema.Num() > 0 && NextAvailableComponentId == SpatialConstants::STARTING_GENERATED_COMPONENT_ID)
+		{
+			UE_LOG(LogSpatialGDKSchemaGenerator, Warning, TEXT("Detected an old schema database, it'll be reset."));
```

Comment:
Isn't this the same as ClearGeneratedSchema?

Reply:
Yes